language/syntaxes/dfdl.tmLanguage.json

```diff
@@ -0,0 +1,387 @@
+				}
+			},
+			"end": "\"",
+			"endCaptures": {
+				"0": {
+					"name": "punctuation.definition.string.end.xml"
+				}
+			},
+			"name": "string.quoted.double.xml",
+			"patterns": [
+				{
+					"include": "#entity"
+				},
+				{
+					"include": "#bare-ampersand"
+				}
+			]
+		},
+		"entity": {
+			"captures": {
+				"1": {
+					"name": "punctuation.definition.constant.xml"
+				},
+				"3": {
+					"name": "punctuation.definition.constant.xml"
+				}
+			},
+			"match": "(&)([:a-zA-Z_][:a-zA-Z0-9_.-]*|#[0-9]+|#x[0-9a-fA-F]+)(;)",
+			"name": "constant.character.entity.xml"
+		},
+		"internalSubset": {
+			"begin": "(\\[)",
+			"captures": {
+				"1": {
+					"name": "punctuation.definition.constant.xml"
+				}
+			},
+			"end": "(\\])",
+			"name": "meta.internalsubset.xml",
+			"patterns": [
+				{
+					"include": "#EntityDecl"
+				},
+				{
+					"include": "#parameterEntity"
+				},
+				{
+					"include": "#comments"
+				}
+			]
+		},
+		"parameterEntity": {
+			"captures": {
+				"1": {
+					"name": "punctuation.definition.constant.xml"
+				},
+				"3": {
+					"name": "punctuation.definition.constant.xml"
+				}
+			},
+			"match": "(%)([:a-zA-Z_][:a-zA-Z0-9_.-]*)(;)",
+			"name": "constant.character.parameter-entity.xml"
+		},
+		"singlequotedString": {
+			"begin": "(?<!\\}\\s*+)'(?!\\s*+\\{)",
+			"beginCaptures": {
+				"0": {
+					"name": "punctuation.definition.string.begin.xml"
+				}
+			},
+			"end": "'",
+			"endCaptures": {
+				"0": {
+					"name": "punctuation.definition.string.end.xml"
+				}
+			},
+			"name": "string.quoted.single.xml",
+			"patterns": [
+				{
+					"include": "#entity"
+				},
+				{
+					"include": "#bare-ampersand"
+				}
+			]
+		},
+		"tagStuff": {
+			"patterns": [
+				{
+					"captures": {
+						"1": {
+							"name": "entity.other.attribute-name.namespace.xml"
+						},
+						"2": {
+							"name": "entity.other.attribute-name.xml"
+						},
+						"3": {
+							"name": "punctuation.separator.namespace.xml"
+						},
+						"4": {
+							"name": "entity.other.attribute-name.localname.xml"
+						}
+					},
+					"match": "(?:^|\\s+)(?:([-\\w.]+)((:)))?([-\\w.:]+)\\s*="
+				},
+				{
+					"include": "#doublequotedString"
+				},
+				{
+					"include": "#singlequotedString"
+				}
+			]
+		},
+		"comments": {
+			"patterns": [
+				{
+					"begin": "<%--",
+					"captures": {
+						"0": {
+							"name": "punctuation.definition.comment.xml"
+						},
+						"end": "--%>",
+						"name": "comment.block.xml"
+					}
+				},
+				{
+					"begin": "<!--",
+					"captures": {
+						"0": {
+							"name": "punctuation.definition.comment.xml"
+						}
+					},
+					"end": "-->",
+					"name": "comment.block.xml",
+					"patterns": [
+						{
+							"begin": "--(?!>)",
+							"captures": {
+								"0": {
+									"name": "invalid.illegal.bad-comments-or-CDATA.xml"
+								}
+							}
+						}
+					]
+				}
+			]
+		}
+	}
+}
```

Comment:
Yes this file is need it defines the syntax for the dfdl language. Such things like comments being `<%-- --%>`, `<!-- -->`  and other just general syntax.

Reply:
But that's not DFDL specific. That's just normal XML/XSD. In fact, the only syntax that is unique to DFDL is values of certain attributes (e.g. DFDL expression). Can we rely on some other extension to define that syntax, and this extension only includes logic for things specific to DFDL (e.g. autocompletion, highlighting expressions) and can build on existing extensions to implement everything else.

Reply:
I mean I think to define the syntax `dfdl` you have to still redefine certain things other languages may use that are similar because I don't think you create a custom language extension and have it import another languages syntax, at least from what I am aware of as the examples I have seen all redefine everything. @rt320 could possibly provide some better insight however

Reply:
Just to add a little more detail to this: you can inherit scopes from other languages - the first implementation of this colorization was using that to effectively add dfdl as an additional alias for XML. The reason we're re-defining the entire XML language in here is because we needed to change an existing setting from the XML. Normally, with inheritance, you would expect that you'd be able to take everything and make the small changes that you need to, but the tmLanguage file doesn't support overwriting like we would need it to.

Reply:
Thanks for the background.

What changes were made to the tmLanguage file? dfdl.tmLanguage.json doesn't seem to have anything specific to DFDL that I can see.

Reply:
Didn't remember what it was, but I found it now - it was the regex for figuring out when strings start/stop. That doesn't make a difference right now, but when I go back and revisit the semantic highlighting changes, we are going to need the changes.

Without the change, the semantic and syntactic highlighting both try to highlight XPath inside of strings. What we want is to turn off the syntactic highlighting for these XPath expressions.